react-sharing-state/my-app/src/Container.js

```diff
@@ -0,0 +1,82 @@
+import React from 'react';
+import { useState } from 'react';
+/**
+ * A container of items.
+ * One item is displayed at a time, with buttons to flip through them:
+ * - Next and Prev scroll forward and backward one item
+ * - An array of buttons scroll to a specific item
+ * TODO: The buttons don't work!
+ */
+export default function Container({ items }) {
+  const [activeIndex, setActiveIndex] = useState(0);
+
+  function handleIt(e) {
+    setActiveIndex(Number(e));
+  }
+
+  return (
+    <div>
+      <div>{items[activeIndex]}</div>
+      <div>
+        <CustomButton
+          text="Prev"
+          onShow={() =>
+            setActiveIndex((activeIndex - 1 + items.length) % items.length)
+          }
+        />
+        <Indicators
+          count={items.length}
+          onSelect={(e) => handleIt(e.target.textContent)}
+          colorDiv={activeIndex}
+        />
+        <CustomButton
+          text="Next"
+          onShow={() => setActiveIndex((activeIndex + 1) % items.length)}
+        />
+      </div>
+    </div>
+  );
+}
+
+/**
+ * A component that wraps a DOM button.
+ * Props:
+ *   text: The button's text
+ *
+ * TODO: Make the background color a prop, default white.
+ * TODO: When clicked, the parent needs to be notified.
+ */
+function CustomButton({ text, color, onShow, isActive }) {
+  color = isActive ? color : 'white';
```

Comment:
That works. It'd be more elegant to use destructuring default values, though. Then you also wouldn't need to pass in `isActive`.